### DISCUSSION TITLE
#32 Band-aid, #37 Tunnel.

### DISCUSSION THREAD

**Discussion by Gabb192** [cssbattle/forum#245] [Q&A]
I was checking my worst targets, two of them are #32 and #37.
What are the ideas behind them? Simply classic css? Unicode? or maybe an appropriate use of box-shadow or something like that?

**Comment by gunnarbittersmann** [comment]
[#32](https://cssbattle.dev/play/32): I can’t speak of @Beowolve’s and @JDMeijers’s solutions, but mine uses Unicode.

[#37](https://cssbattle.dev/play/37) uses nothing that @rafehimself would complain about yet. Which might change with the next server update.

**Reply by Gabb192** [reply]
Ok, thank you.

**Reply by Beowolve** [reply]
#32 is Unicode for me as well

**Comment by Gabb192** [comment]
Talking about #86 it is just clip-path? Because I managed to get to 139, but I don't think that only with path I can squeeze more.

**Reply by SelenIT** [reply]
My solution is `clip-path`, I guess it can be squeezed a couple of characters more. My best without clip-path was 149.

**Reply by Gabb192** [reply]
Thanks!

**Reply by verberg** [reply]
Is there any clip-path generator?

**Reply by Beowolve** [reply]
I recommend:
https://github.com/Yqnn/svg-path-editor

**Reply by verberg** [reply]
If I have width this generator a path, for example: M0 0 10 1 8 3 3 1l-3 4L0 0 (absolute path) or m0 0 10 1-2 2-5-2-3 4 0-5 (relative path). How can I put it into the short-form like "clip-path:pat\68('..... ') like the top-solution in battle #100?

Or as an other question: If I have a path in the "normal way" of clip-path:polygon(.....) - how to transform this code in the "short-form" like in battle #100?

**Reply by rafehimself** [reply]
@verberg Hey Verena, for your second question, I don't think that's possible in a smooth way because the syntax for both path is different css units like `clip-path:polygon(0 0, 20q 20q, 0 40q)`
 vs svg syntax what you wrote in you first question, correct me if I'm wrong fellows 🥲 ,
however with that [path-genraotr](https://github.com/Yqnn/svg-path-editor) it's quite easy,  with a bit of practice you can make for example target #4 in less than 10 minutes. It' less like coding more like photoshop 🤣 
I'm not sure what you are asking for in first question, you put that path into `clip-path:pat\68('M0 0 10 1 8 3 3 1l-3 4L0 0')` and check which values are shorter, usually relative, but not always! also you need something from which that path will be cut

**Reply by verberg** [reply]
@rafehimself: Thanks a lot! Now - two or three or four hours later - I think I understand much more about this paths and at target #4 I broke my own record, I have now 146 chars!

**Reply by verberg** [reply]
@rafehimself .. and now 136 chars!

**Reply by Beowolve** [reply]
130 is the current record for this target with clip-path

**Reply by verberg** [reply]
@beowolf : Then better results are with Unicode?

**Reply by Beowolve** [reply]
@verberg You referenced a other beo.... ;). And yes the top solutions are unicode based, there are some very good characters to create that. Basically u only need boxes and circles.

**Reply by SelenIT** [reply]
> Or as an other question: If I have a path in the "normal way" of clip-path:polygon(.....) - how to transform this code in the "short-form" like in battle #100?

Apparently if the coordinates of the polygon are set in pixels (e.g.`polygon(0 5px,55px 70px,...`), these coordinates can be directly translated into the `path()` syntax as absolute coordinates of the "moveTo" and "lineTo" commands, i.e. `M` for the starting point and 'L' (optional, implied if missing) for other points: `path('M0 5 L55 70 ...')`. After that, this path can be run through some tool like SVGO to remove redundant commands and spaces, convert absolute coords to relative etc.

> Then better results are with Unicode?

121 is still possible with borders and a bit of clipping (although the more precise 122-char version is not anymore, ironically) :)

**Comment by midnightPush** [comment]
and if I may ask, how #88 is done with so little characters ? 👀

**Reply by Beowolve** [reply]
unicode

**Reply by joewinbpaul** [reply]
> unicode

Ummm... can u explain exactly how?   Or show the Unicode chars u used 😂🤣

**Reply by Gabb192** [reply]
I can't show you the unicode chars.. but for sure two rounded bars are needed, and for the middle part you need to be a little creative. (You choose one of the hardest target to do with unicode due to the difficulty of finding the right characters and the difficulty of making 100%) @joewinbpaul

**Reply by joewinbpaul** [reply]
> I can't show you the unicode chars.. but for sure two rounded bars are needed, and for the middle part you need to be a little creative. (You choose one of the hardest target to do with unicode due to the difficulty of finding the right characters and the difficulty of making 100%) @joewinbpaul

Thx even tho it didn't increase my score as I don't know about any rounded Unicode bars...

**Reply by joewinbpaul** [reply]
> > I can't show you the unicode chars.. but for sure two rounded bars are needed, and for the middle part you need to be a little creative. (You choose one of the hardest target to do with unicode due to the difficulty of finding the right characters and the difficulty of making 100%) @joewinbpaul
> 
> Thx even tho it didn't increase my score as I don't know about any rounded Unicode bars...

I managed to find _one_ Unicode char which, _`after experimenting with a little bit`_, got me a 99.9% match with 158chars <p> I have no idea if I even got the right unicode char
